src/test/java/org/jenkinsci/plugins/gitclient/GitAPITestUpdate.java

```diff
@@ -439,6 +438,58 @@ public void tearDown() throws Exception {
         }
     }
 
+    private boolean fileProtocolIsAllowed = false;
+
+    /**
+     * Allow local git clones to use the file:// protocol by setting
+     * protocol.file.allow=always in the global git configuration for
+     * the current user.
+     *
+     * Command line git 2.38.1 and patches to earlier versions
+     * disallow local git submodule cloning with the file:// protocol.
+     * The change resolves a security issue but that security issue is
+     * not a threat to these tests.
+     *
+     * After the test completes, the setting is removed from the
+     * global git configuration of the current user.
+     *
+     * The setting must be applied to the global configuration of the
+     * current user or the affected tests will fail. It is unfortunate
+     * that the tests are modifying the global configuration of the
+     * current user, but it is more important that the tests be run
+     * than that we defend the git settings of git plugin developers
+     * from changes performed by the tests.
+     */
+    protected void allowFileProtocol() throws Exception {
+        w.launchCommand("git", "config", "--global", "protocol.file.allow", "always");
+        fileProtocolIsAllowed = true;
+    }
+
+    /**
+     * Removes the protocol.file.allow setting from the global git
```

Comment:
What if the developer’s Git config deliberately included this flag before running tests?

Reply:
This will remove the flag even if it was previously set.  I didn't feel that was a critical condition to preserve in the developer's git config.  If you feel that it is, I'm happy to investigate some way of detecting the current state and preserving the current state.  I've not been very successful with those techniques in the past, but I'm willing to try again if it is a serious concern.

Reply:
Resolved by switching from a change of global configuration to changing the arguments to command line git.  Thanks for the suggestion!